src/go/cmd/http-relay-client/main.go

```diff
@@ -449,10 +511,16 @@ func handleRequest(remote *http.Client, local *http.Client, req *pb.HttpRequest)
 					duration := time.Since(ts)
 					resp.BackendDurationMs = proto.Int64(duration.Milliseconds())
 					// see makeBackendRequest()
-					urlPath := strings.TrimPrefix(*req.Url, "http://invalid")
+					urlPath := strings.TrimPrefix(*pbreq.Url, "http://invalid")
 					log.Printf("[%s] Backend request duration: %.3fs (for %s)", *resp.Id, duration.Seconds(), urlPath)
+				} else {
+					// Q(hauke): When are we ending up in this branch?
+					// What are the semantics and why are we not setting a request duration?
+					// Even in a streaming case I would expect a duration which represents the
+					// processing time of the last item.
```

Comment:
we could track each backend request and report the progress. let try as a followup